scrabble.js

```diff
@@ -1,8 +1,187 @@
-var Scrabble = function() {};
+var Scrabble = {};
 
 // YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
+Scrabble.helloWorld = function() {
   return 'hello world!';
 };
 
+Scrabble.score = function(word) {
+  word = word.toLowerCase();
+  var total = 0;
+  var scoreChart = {
+    "aeioulnrst": 1,
+    "dg": 2,
+    "bcmp": 3,
+    "fhvwy": 4,
+    "k": 5,
+    "jx": 8,
+    "qz": 10
+  };
+
+
+  // Loop over each letter in word, checking against each score key
+  for(var i = 0; i < word.length; i++){
+    for (var key in scoreChart) {
+      if(key.indexOf(word[i]) >= 0) {
+        total += scoreChart[key];
+      }
+    }
+  }
+  return total;
+};
+
+Scrabble.highestScoreFrom = function(arrayOfWords) {
+
+  var scores = {};
+
+  // Assign scores to each word
+  for (var i = 0; i < arrayOfWords.length; i++) {
+    scores[arrayOfWords[i]] = this.score(arrayOfWords[i]);
+  }
+
+  // get largest score regardless of duplicates
+  var maxScore = Math.max.apply(null, Object.values(scores));
+
+  // get all words with the largest score
+  var potentialWinners = [];
+  for (var word in scores) {
+    if (scores[word] == maxScore) {
+      potentialWinners.push(word);
+    }
+  }
+
+  // go through winners array
+  if (potentialWinners.length == 1) {
+    return potentialWinners[0];
+  }
+  else {
+    // return seven-letter word if there is one (we assume there's only one...)
+    console.log(potentialWinners);
+    for (var j = 0; j < potentialWinners.length; j++) {
+      console.log(potentialWinners[j].length);
+      if (potentialWinners[j].length === 7) {
+        return potentialWinners[j];
+      }
+    }
+    // return shortest word
+    var min = potentialWinners[0].length;
+    var highestWord = potentialWinners[0];
+    for (var k = 0; k < potentialWinners.length; k ++) {
```

Comment:
You should be able to use the same `i` iteration variable over again since these are independent. This is a pretty common situation for independent loops like this.